Azure.ResourceManager.Core/Adapters/ModelAdapters.cs

```diff
@@ -17,6 +17,20 @@ protected TrackedResource(ResourceIdentifier id, Location location, T data)
             Model = data;
         }
 
+        protected TrackedResource(string id, Location location, T data)
```

Comment:
need some context as to why this is needed.  Is this to avoid the automatic implicit call, allowing us to have null as the id for "request" objects?

Reply:
Exactly, if null get's passed into the other constructor, the implicit this get's called. This allows us to properly avoid that and set null for request types